File: admin-ui/src/public-point/publicPointService.js
import restApi, { buildGetFilterParams } from '../common/restApi';

const BASE_URL = '/pp-service/public-points';

const publicPointService = {
    find: function (filter, pageable, sort) {
        return restApi.get(BASE_URL, {
            params: buildGetFilterParams(filter, pageable, sort)
        });
    },
    findById: function (id) {
        return restApi.get(`${BASE_URL}/${id}`);
    },
    findFullDetailsById: function (id) {
        return restApi.get(`${BASE_URL}/${id}/full-details`);
    },
    update: function (id, request) {
        return restApi.put(`${BASE_URL}/${id}`, request);
    },
    create: function (request) {
        return restApi.post(BASE_URL, request);
    },
    findNames: function (companyId) {
        return restApi.get(`${BASE_URL}/names`, {
            params: { companyId }
        });
    },
    changeStatus: function (id, request) {
        return restApi.put(`${BASE_URL}/${id}/status`, request);
    }
}

export default publicPointService;

File: pp-service/src/main/java/com/acme/ppsrv/publicpoint/service/PublicPointService.java
package com.acme.ppsrv.publicpoint.service;

import com.acme.commons.security.UserAuthenticated;
import com.acme.ppsrv.publicpoint.PublicPointStatus;
import com.acme.ppsrv.publicpoint.dto.CreatePublicPointDto;
import com.acme.ppsrv.publicpoint.dto.FullDetailsPublicPointDto;
import com.acme.ppsrv.publicpoint.dto.PublicPointDto;
import com.acme.ppsrv.publicpoint.dto.PublicPointFilter;
import com.acme.ppsrv.publicpoint.dto.UpdatePublicPointDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.validation.annotation.Validated;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import java.util.Collection;
import java.util.UUID;

@Validated
public interface PublicPointService {
    @PreAuthorize("hasAnyAuthority('ADMIN','COMPANY_OWNER')")
    Mono<UUID> create(@Valid CreatePublicPointDto dto);

    @PreAuthorize("hasAnyAuthority('ADMIN','COMPANY_OWNER')")
    Mono<Void> update(UUID id, @Valid UpdatePublicPointDto dto);

    @PreAuthorize("hasAnyAuthority('ADMIN','COMPANY_OWNER')")
    Mono<Void> changeStatus(UUID id, @NotNull PublicPointStatus newStatus);

    @UserAuthenticated
    Mono<PublicPointDto> findById(UUID id);

    @PreAuthorize("hasAnyAuthority('ADMIN','COMPANY_OWNER')")
    Mono<FullDetailsPublicPointDto> findFullDetailsById(UUID id);

    @PreAuthorize("hasAnyAuthority('ADMIN','COMPANY_OWNER')")
    Mono<Page<FullDetailsPublicPointDto>> find(PublicPointFilter filter, Pageable pageable);

    @PreAuthorize("hasAnyAuthority('ADMIN, COMPANY_OWNER')")
    Flux<PublicPointDto> findNames(UUID companyId);

}
